common/events.d.ts

```diff
@@ -18,10 +18,51 @@ interface MoveCameraEvent {
     lerp: number;
 }
 
+interface DialogConfirmedEvent {
+    PlayerID: PlayerID;
+}
+
+interface DialogReceivedEvent {
+    DialogText: string;
+    SendToAll: boolean;
+    DialogEntIndex: EntityIndex;
+    PlayerHeroEntIndex: EntityIndex;
+    ShowAdvanceButton: number;
+    DialogLine: number;
+    ConfirmToken: string;
+    DialogPlayerConfirm: boolean | number;
+    DialogAdvanceTime: number;
+}
+
+interface DialogCompleteEvent {
+    DialogEntIndex: EntityIndex | null;
+    DialogLine: number;
+    ShowNextLine: number;
+    PlayerHeroEntIndex: EntityIndex;
+}
+
+interface DialogConfirmEvent {
+    nPlayerID: PlayerID;
+    ConfirmToken: string | null;
+    DialogEntIndex: EntityIndex | null;
+    DialogLine: number;
+}
+
+interface DialogConfirmExpireEvent {
+    ConfirmToken: string | null;
+    DialogEntIndex: EntityIndex | null;
+    DialogLine: number;
+}
+
 interface CustomGameEventDeclarations {
     section_started: SectionStartedEvent;
     skip_to_section: SkipToSectionEvent;
     move_camera: MoveCameraEvent;
+    dialog_confirm_expire: DialogConfirmExpireEvent;
+    dialog_confirm: DialogConfirmEvent;
+    dialog_complete: DialogCompleteEvent;
+    dialog: any; // DialogReceivedEvent, but doesn't work because I don't understand typescript
```

Comment:
I gave it a go and think I see what the problem is: You try to directly handle the event with this data in dialog.ts line 14, but actually if you look at the type of Subscribe, the type of the event arriving in the client is not `DialogReceivedEvent`, but `NetworkedData<DialogReceivedEvent>`. This is because while sending data form lua to panorama, booleans are transformed to 0/1 and arrays are sent as objects with string keys (see questpanel.ts/parseServerArray).